lib/arproxy/base.rb

```diff
@@ -2,8 +2,8 @@ module Arproxy
   class Base
     attr_accessor :proxy_chain, :next_proxy
 
-    def execute(sql, name=nil, **kwargs)
```

Comment:
@nekketsuuu Good question.
Indeed, in the current Arproxy, `Arproxy::Base#execute` actually replaced the call to ConnectionAdapter’s `#execute`, so `kwargs` was necessary as an argument.
However, in Arproxy v1, a new class named `Arproxy::ConnectionAdapterPatch` takes on this role. `Arproxy::Base#execute` is called with two arguments of `(sql, name)` from the patch, so `kwargs` is no longer needed in this context.

https://github.com/cookpad/arproxy/pull/32/files#diff-b0f45f0d5c3f0e5178dce37a7f81a3b41699ebec5c4d0a2a3cff7d5a53b7dbcaR72

Reply:
I agree that this won't break ConnectionAdapter's `execute`, but I mean that this may break the behavior of `execute` of the existing proxies (classes that inherit `Arproxy::Base`). If these proxies use the given keyword arguments, all of them will be changed to `nil` in v1. I heard that you'd like to maintain backward compatibility for proxies (<https://github.com/cookpad/arproxy/issues/30> ), so I wonder if this is your intention.

Reply:
I'm still struggling with this change in specification, but in order to keep up with ActiveRecord version upgrades, I'd like to release it with this specification first.
I've also written about this non-compatible change in the README.

https://github.com/cookpad/arproxy/tree/v1?tab=readme-ov-file#upgrading-guide-from-v0x-to-v1